src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -50,9 +60,18 @@ public Integer countInnReachLoans(String patronId, List<UUID> loanIds) {
 
   @Override
   public InnReachTransactionDTO getInnReachTransaction(UUID transactionId) {
-    return repository.fetchOneById(transactionId)
-      .map(transactionMapper::toDTO)
+    var transaction = repository.fetchOneById(transactionId)
       .orElseThrow(() -> new EntityNotFoundException(String.format("InnReach transaction with id [%s] not found!", transactionId)));
+    if (transaction.getState() == ITEM_SHIPPED || transaction.getState() == ITEM_RECEIVED || transaction.getState() == RECEIVE_UNANNOUNCED) {
+      transaction.setState(RECALL);
+      log.info("Get innReach transaction with id {} and {} status", transaction.getId(), transaction.getState());
+      var loan = loanService.getById(transaction.getHold().getFolioLoanId());
+      var loanDueDate = loan.getDueDate().toInstant().truncatedTo(ChronoUnit.SECONDS);
+      var loanIntegerDueDate = (int) (loanDueDate.getEpochSecond());
+      transaction.getHold().setDueDateTime(loanIntegerDueDate);
+      notifier.reportRecallRequested(transaction, loanDueDate);
+    }
+    return transactionMapper.toDTO(transaction);
```

Comment:
I suppose, this logic should be added to a new method in InnReachTransactionActionService rather then implemented here, because this method is used for a simple retrieving of a transaction and if you add this logic here it will recall the transaction each time transaction is retrieved.

Reply:
Yes i know